src/Sarif.Converters/TextFormats/CsvReader.cs

```diff
@@ -0,0 +1,148 @@
+{
+    /// <summary>
+    ///  CsvReader provides a simple interface for reading CSV files.
+    ///  
+    ///  Usage:
+    ///  using(CsvReader reader = new CsvReader("Input.csv"))
+    ///  {
+    ///      while(reader.NextRow())
+    ///      {
+    ///           List&lt;string&rt; row = reader.Current();
+    ///           ...
+    ///      }
+    ///  }
+    /// </summary>
+    /// <remarks>
+    ///  Loosely based on: https://github.com/Microsoft/elfie-arriba/blob/master/Elfie/Elfie/Serialization/CsvReader.cs
+    /// </remarks>
+    internal class CsvReader : BaseTabularReader
+    {
+        public CsvReader(string filePath, int bufferSize = DefaultBufferSize) : this(File.OpenRead(filePath), bufferSize)
+        { }
+
+        public CsvReader(Stream stream, int bufferSize = DefaultBufferSize) : base(stream, bufferSize)
+        { }
+
+        public override bool NextRow()
+        {
+            bool fileHasMore = true;
+
+            // Read more content if needed, returning false without incrementing RowCountRead if done
+            if (_nextIndex >= _bufferFilledLength)
+            {
+                fileHasMore = RefillBuffer();
+                if (!fileHasMore) { return false; }
+            }
+
+            _currentRow.Clear();
+            RowCountRead++;
+
+            while (true)
+            {
+                int cellEnd = _nextIndex;
+                bool quoted = (cellEnd < _bufferFilledLength && _buffer[cellEnd] == '"');
+
+                // Find the end of the current cell
+                cellEnd = NextCellEnd(quoted, out bool escapedQuotes);
+
+                // If we didn't find a delimiter and there might be more content, read more and try again
+                if (cellEnd >= _bufferFilledLength && fileHasMore)
+                {
+                    fileHasMore = RefillBuffer();
+                    continue;
+                }
+
+                // If we found a delimiter or end of file, read in the cell value
+                string value;
+                if (quoted)
+                {
+                    value = new string(_buffer, _nextIndex + 1, cellEnd - _nextIndex - 2);
+                    if (escapedQuotes) { value = value.Replace("\"\"", "\""); }
+                }
+                else
+                {
+                    value = new string(_buffer, _nextIndex, cellEnd - _nextIndex);
+                }
+                _currentRow.Add(value);
+                _nextIndex = cellEnd + 1;
+
+                // If the last delimiter was not a cell delimiter, we finished reading the row
+                if (!(cellEnd < _bufferFilledLength && _buffer[cellEnd] == ','))
+                {
+                    break;
+                }
+            }
+
+            // Skip newline for \r\n to set _nextIndex to the beginning of the next line
+            if (_nextIndex <= _bufferFilledLength && _buffer[_nextIndex - 1] == '\r') { _nextIndex++; }
```

Comment:
>'\r' [](start = 80, length = 4)

Shouldn't we be looking for "\r\n" or "\n" here, in case the file was created with unix line endings?

Reply:
It handles both styles. The logic earlier to look for separators looks for ',', '\r', and '\n'. This line skips over the \n also if we got \r as the separator, since the next value won't start until after both parts of Windows style newlines.

Unit tests cover both newline styles (even mixed in one file)